### ISSUE TITLE
java.lang.NoClassDefFoundError: com.pusher.client.-$$Lambda$Pusher$roDLx-G6z2LMO45pVffrVWTG41U

### ISSUE THREAD

**Issue by bettkipronoh** [pusher/pusher-websocket-java#335]
### What is the issue?
The app crashes when using v 2.4.0 at runtime giving the following error. If i use 2.2.8, it seems to work well.
`java.lang.NoClassDefFoundError: com.pusher.client.-$$Lambda$Pusher$roDLx-G6z2LMO45pVffrVWTG41U
W/System.err:     at com.pusher.client.Pusher.<init>(Pusher.java:108)
W/System.err:     at com.pusher.client.Pusher.<init>(Pusher.java:89)
W/System.err:     at com.***********.********Activity.connectPusher`

The error occurs when calling this 
`pusher = new Pusher("*********************", options);`

...

---- 

CC @pusher/mobile

**Comment by benw-pusher** [comment]
I am not able to replicate this, are you able to share your code regarding this?

**Comment by Abdallah-Abdelazim** [comment]
Refer to this issue #334

**Comment by Abdallah-Abdelazim** [comment]
@benw-pusher to reproduce this issue run the Android app on a device running Android Marshmallow or below.

**Comment by Abdallah-Abdelazim** [comment]
@benw-pusher Yes, I have.
I work on an app that has 200k+ active users. After we upgraded the Pusher library from version `2.2.1` to version `2.4.0`. We noticed a lot of crashes reported to Crashlytics as `Fatal Exception: java.lang.NoClassDefFoundError` in `Pusher.java line 108` which used `java.util.function.BiConsumer`. The crashes were only on Android 5 & 6 (our minSdk is 21 i.e. Android 5). I was able to reproduce the crashes by running the app on Android 5 emulators.
I searched a bit and found this [link](https://developer.android.com/studio/write/java8-support#library-desugaring) that mentions some Java APIs such as `java.util.function` to require *API desugaring* to work on older Android versions.
Here's the code snippet from the link I mentioned earlier that helped me fix this crash:
```groovy
android {
    defaultConfig {
        // Required when setting minSdkVersion to 20 or lower
        multiDexEnabled true
    }

    compileOptions {
        // Flag to enable support for the new language APIs
        coreLibraryDesugaringEnabled true
        // Sets Java compatibility to Java 8
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
}

dependencies {
    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'
}
```

I think this repo should mention this in its README.
